runtimes/protocol/lsp.ts

```diff
@@ -85,6 +85,13 @@ export interface AWSInitializationOptions {
      * Additional clientInfo to extend or override default data passed by LSP Client.
      */
     clientInfo?: ExtendedClientInfo
+    /**
+     * Client data directory, used as an application data directory to store LSP server-specific data.
+     * LSP servers are expected to use the directory for storing their configuration files,
+     * temporary data, and user-specific settings.
+     * The directory is extected to maintain data across updates and re-installations.
+     */
+    dataDirectory?: string
```

Comment:
"Data" is a bit nebulous, is there a crisper name for this parameter?  Something like `lspHome` or `serverFolder`?

nit:  In searching the LSP spec, "folder" is the terminology used in the spec rather than "directory".  Consider renaming for consistency.

Is the structure under that folder covered somewhere else?  i.e. What will enforce each LSP server to isolate its data in a sane and consistent manner (not just start dumping generically named files under the folder path provided)?  

What happens if multiple IDEs use the same path (say VSC and JetBrains installed on the same machine and both pass `$HOME/awsLanguageServers` or something like that)?  

Is the expectation that a configuration or data change in one IDE would impact all other tools consuming the language servers on that host?

Reply:
Will rename in followup PR. Also, will extend comment with more details. I think the recommendation for the client would be to provide folder that is specific per-client/IDE.

Reply:
About LSP server isolating its data within the folder - we can think about exposing `getClientDataDirPath` from `fs` feature as an abstraction to facilitate proper usage of the folder.

Reply:
Also, keep in mind that this folder is optional in parameters. I introduced it for cases like [globalStorageUri](https://code.visualstudio.com/api/extension-capabilities/common-capabilities#data-storage) in vscode. If not specified, runtime will define it - with subfolder per server per client.